tests/framework/test_engine.py

```diff
@@ -25,6 +26,12 @@ def is_same_object_method(m1, m2):
     return m1.__func__ is m2.__func__ and m1.__self__ is m2.__self__
 
 
+@pytest.fixture()
+def SimulationContext():
+    yield SimulationContext_
+    SimulationContext_._clear_context_cache()
```

Comment:
Clever use of yield.

Reply:
When does `SimulationContext_._clear_context_cache()` get triggered? I think it'd be helpful to add a comment explaining exactly what is happening here.

Reply:
Disagree.  This is the standard teardown procedure for fixtures: https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended